Cutesv

yuukiiwa opened the pull request:
<!--
# nf-core/modules pull request

Many thanks for contributing to nf-core/modules!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the master branch.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

Closes #XXX <!-- If this PR fixes an issue, please link it here! -->

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

nvnieuwk reviewed (commented):
Mostly the same things as for Sniffles, can you try and fix the module to the guideline suggestions I did there?

yuukiiwa commented:
Thanks for reviewing, @nvnieuwk! I have updated the module based on the suggestions you had for `sniffles`. Here are the changes I have made:
1. using `tuple val(meta2), path(fasta)` in `modules/nf-core/cutesv/main.nf` and updated `modules/nf-core/cutesv/meta.yml` and `test/modules/nf-core/cutesv/main.nf` accordingly
2. using `vcf` instead of `sv_vcf` for the output channel and removed the custom suffices from the output `vcf` file
3. using `def prefix = task.ext.prefix ?: "${meta.id]"` for output names
4. added `cutesv` to `test/config/pytest_modules.yml`

nvnieuwk commented on modules/nf-core/cutesv/main.nf:
```diff
@@ -0,0 +1,40 @@
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def genotyping = params.enable_genotyping ? "--genotyping" : ''
```
You can supply this flag using `task.ext.args` no need to make a separate argument for this

yuukiiwa replied:
I have removed this

nvnieuwk commented on modules/nf-core/cutesv/main.nf:
```diff
@@ -0,0 +1,40 @@
+    cuteSV \
+        ${bam} \\
+        ${fasta} \\
+        ${prefix}.vcf \\
```
Does this tool output bgzipped output? If so, having bgzipped output is preffered

yuukiiwa replied:
for outputting a `.bgzip` file would require an additional command
```
bgzip -c tools/cutesv/cutesv.vcf > tools/cutesv/cutesv.vcf.gz
```

nvnieuwk replied:
That would be perfect (if `bgzip` is in the containers for CuteSV, otherwise it's fine to leave it unzipped)

yuukiiwa replied:
`bgzip` is not in the container

nvnieuwk replied:
Ahh too bad, it's fine then :)

yuukiiwa replied:
Hi @nvnieuwk, can you approve the PR, please?

yuukiiwa commented on modules/nf-core/cutesv/main.nf:
```diff
@@ -0,0 +1,40 @@
+        ${prefix}.vcf \\
+        . \\
+        --threads $task.cpus \\
+        --sample ${meta.id} \\
```
I have just tried running after removing `--sample`, which is fine, so I don't think `--sample` is mandatory

nvnieuwk replied:
It should be in `task.ext.args` then according to the [guidelines](https://nf-co.re/docs/contributing/modules#general)

nvnieuwk commented on modules/nf-core/cutesv/meta.yml:
```diff
@@ -0,0 +1,55 @@
+        e.g. [ id:'test' ]
+  - vcf:
+      type: file
+      description: Compressed VCF file
+      pattern: "*.vcf.gz"
```
```suggestion
      description: VCF file containing called variants from CuteSV
      pattern: "*.vcf"
```
